src/devices/machine/cs8900a.cpp

```diff
@@ -0,0 +1,1522 @@
+// license:GPL-2.0+
+// copyright-holders:Rhett Aultman
+/*************************************************************************
+
+    CS8900A ethernet controller implementation
+
+    by Rhett Aultman <roadriverrail@gmail.com>
+    ported to MAME from VICE Project (https://sourceforge.net/p/vice-emu/)
+    VICE CS8900 code by Spiro Trikaliotis <Spiro.Trikaliotis@gmx.de>
+
+**************************************************************************/
+
+#include <cstring>
+
+#include "machine/cs8900a.h"
+
+DEFINE_DEVICE_TYPE(CS8900A,  cs8900a_device,  "CS8900A",  "CA8900A ETHERNET IC")
+
+/* warn illegal behaviour */
+/* #define CS8900_DEBUG_WARN_REG 1 */   /* warn about invalid register accesses */
+/* #define CS8900_DEBUG_WARN_RXTX 1 */  /* warn about invalid rx or tx conditions */
+
+/** #define CS8900_DEBUG 1 **/           /* enable to see port reads */
+/** #define CS8900_DEBUG_INIT 1 **/
+/** #define CS8900_DEBUG_LOAD 1  **/           /* enable to see port reads */
+/** #define CS8900_DEBUG_STORE 1  **/          /* enable to see port writes */
+/** #define CS8900_DEBUG_REGISTERS 1 **/      /* enable to see CS8900a register I/O */
+/** #define CS8900_DEBUG_IGNORE_RXEVENT 1 **/ /* enable to ignore RXEVENT in DEBUG_REGISTERS */
+/** #define CS8900_DEBUG_RXTX_STATE 1 **/     /* enable to see tranceiver state changes */
+/** #define CS8900_DEBUG_RXTX_DATA 1 **/      /* enable to see data in/out flow */
+/** #define CS8900_DEBUG_FRAMES 1 **/         /* enable to see arch frame send/recv */
+
+#define log_message(level,...) fprintf(stderr,__VA_ARGS__)
```

Comment:
Suggest #including "logmacro.h" and using the LOG_X and VERBOSE macros instead.

Reply:
Yes, see examples in other files that use `logmacro.h` – it’s better if this stuff is done in a consistent way.  Also avoid using `printf` family functions in general, they behave differently on different targets and are not type-safe.  It’s better to use `logerror` and MAME’s `osd_printf_` family of functions,

Reply:
deb674f74309b64f2a06d72841075c5ab9ff40eb is a recent example of using `logmacro.h`